### PR TITLE
Add "I'm feeling lucky" language search (#392)

### LINKED ISSUE #392: Add "I'm feeling lucky" language search
Right now when people in the intro page type in a Language and click GO it opens a card list with the potential languages and a limit of 1 -- however we changed how details works and now it just looks funny. We should remove the limit of 1 parameter and instead add a route that receives the query and automatically sets the objectID parameter to the first result, but showing other search results if they exist.

You'll want to make a new page `LuckySearchPage.tsx` You can start this page by copying `DataPage.tsx` since we are using params & the data layer -- but you can leave out `HoverCardProvider` but keep the `Suspense` in since there will be a delay between the page is opened and the code is loaded. For the page body you should make  `LuckySearchPageBody.tsx` -- where the result value will just be "Searching...". The real mechanism that you will be using in the Page Body is code living in a useEffect clause that will wait for the data to load and it will search languages.

In your component you will load data from various sources. If you successfully wrapped the body in the Data & Param providers you should have no problem re-using components from the Data page:
```ts
const {searchString } = usePageParams();
const {getLanguage, languagesInSelectedSource} = useDataContext();
const {filteredObjects} = useFilteredObjects();

useEffect(() => {
   ...
}, [searchString, getLanguage, languagesInSelectedSource, filteredObjects]);
```

Procedure for the useEffect clause:
1. Stop if the length of languagesInSelectedSource is 0 (it hasn't loaded yet)
2. Check if `getLanguage(searchString)` returns a result, if so, go to the page passing the searchString and object ID.
   1. I think that redirect looks like `redirect('data?searchString=' + searchString + '&objectID=' + objectID);`
4. Use `useFilteredObjects` -- this will use the same filters that the CardList view uses
5. If that array has at least 1 result, go to the DataPage with params `{searchString, objectID: filteredObjects[0].ID}`6. If it doesn't, I leave it to you to decide the best back-up UX to show the user, maybe an adaptation of `SearchBar.tsx`

In order to access this URL you will add a row for it in `PageRoutes.tsx`. The page will not appear in the navbar though since it's only a utility page. When you add it to the routes you can test it yourself by going to `.../lang-nav/lucky?searchString=Chinese`. You will plug this in by updating `CommonObjectives` and point the first objective to 1) this page and 2) don't set the limit to 1.

|Step|Screenshot|
|--|--|
|Type in language in [intro page](https://translation-commons.github.io/lang-nav/intro) |Image
|Currently: Goes to a card list with 1 result [link](https://translation-commons.github.io/lang-nav/data?searchString=Chinese&limit=1)|Image
|Proposed: Goes to regular card list with filter and the first result is pre-picked to see in Details [link](https://translation-commons.github.io/lang-nav/data?searchString=Chinese&objectID=zho)|Image

### PR DESCRIPTION
**Pull request by hdprajwal**
Fixes #392 

Summary: Implemented an "I'm feeling lucky" search feature that automatically redirects users to the most relevant language result. When users search for a language on the intro page, they now see all matching results with the first one pre-selected in the details panel, instead of showing a card list limited to 1 result.

### Changes

- User experience
  - Added `/lucky` route that performs search and redirects to best match
  - Updated "Find information about a language" objective on intro page to use lucky search
  - Users now see full search results with first result auto-selected in details panel
  - Added fallback UI with SearchBar when no results are found
- Logical changes
  - Created `LuckySearchPage` and `LuckySearchPageBody` components
  - Implements 3-tier search logic: (1) exact ID match via `getLanguage()`, (2) filtered search via `useFilteredObjects()`, (3) no results fallback
  - Redirects to `/data?searchString={query}&objectID={firstResult.ID}` when match found
- Data
  - No data changes
- Refactors
    - Refactored `Objective` component to accept optional `urlPath` prop (defaults to `'data'`)
    - Updated `GoButton` to use dynamic `urlPath` instead of hardcoded `'data'`

Out of scopes/Future work:
- Could potentially add "Did you mean?" suggestions for close matches when search fails.

### Test Plan and Screenshots

How to test the changes in this PR:
  1. Go to http://localhost:5173/lang-nav/intro
  2. Type "Chinese" in the first objective input field
  3. Click the GO button
  4. Verify it redirects to `/lucky?searchString=Chinese` then to `/data?searchString=Chinese&objectID=zho`
  5. Verify the details panel shows Chinese language information
  6. Test edge cases: exact ID (`zho`), partial match (`Span`), no results (`XYZ123`)

| Page/View with link | Description of Changes | Screenshot Before | Screenshot After |
| ------------------- | ---------------------- | ----------------- | ---------------- |
| [Intro page](http://localhost:5173/lang-nav/intro) => Search "Chinese" | Changed from redirecting to `/data?limit=1` to `/lucky` route that finds best match | image | image |
| Lucky page with no results | New fallback UI when no matches found | N/A (didn't exist before) | image |

## Manual testing results
- [x] Direct ID match (`zho`) → Redirects correctly
- [x] Name search (`Chinese`) → Finds and redirects to first match
- [x] Partial match (`Span`) → Redirects to Spanish
- [x] No results (`XYZ123`) → Shows fallback UI with SearchBar
- [x] From intro page → Full flow works as expected

# Checklist

## Summary

- [x] Clear description of what and why

## Testing

- [x] `npm run lint`
- [x] `npm run build`
- [x] `npm run test`
- [x] `npm run dev` -- tried out the website directly
  - [x] Include screenshots as noted below
  - [x] Write comments on manual testing

## Docs

- [x] Code is self-documenting, or if not, comments are added where needed.

### CODE REVIEW

File: src/widgets/CommonObjectives.tsx [hunk 1]
```diff
@@ -46,14 +46,16 @@ type ObjectiveProps = {
   label: string;
   inputPlaceholder?: string;
   inputParam?: keyof PageParamsOptional;
-  urlParams: PageParamsOptional;
+  urlParams?: PageParamsOptional;
+  urlPath?: string;
```
Comment (conradarcturus) [inline src/widgets/CommonObjectives.tsx]: Nit: use the enum. Generally I avoids string types unless the text is truly freeform.
```suggestion
  urlPath?: LangNavPageName;
```

**Review by conradarcturus — APPROVED** [review]
Looks great! Works great!

I have some style feedback but that's minor.
